tests/unit/models/keras/test_sampler.py

```diff
@@ -14,75 +14,222 @@
 
 _ENSEMBLE_SIZE = 3
 
 
-@pytest.mark.parametrize("num_evals", [10, 20])
+@pytest.fixture(name="diversify", params=[True, False])
+def _diversify_fixture(request: Any) -> bool:
+    return request.param
+
+
+@pytest.fixture(name="num_evals", params=[9, 19])
+def _num_evals_fixture(request: Any) -> int:
+    return request.param
+
+
+@pytest.fixture(name="batch_size", params=[1, 2])
+def _batch_size_fixture(request: Any) -> int:
+    return request.param
+
+
 def test_ensemble_trajectory_sampler_returns_trajectory_function_with_correctly_shaped_output(
     num_evals: int,
+    batch_size: int,
+    dim: int,
+    diversify: bool,
 ) -> None:
-    example_data = empty_dataset([1], [1])
-    test_data = tf.linspace([-10.0], [10.0], num_evals)
-    test_data = tf.expand_dims(test_data, -2)  # [N, 1, d]
+    """
+    Inputs should be [N,B,d] while output should be [N,B]
+    """
+    example_data = empty_dataset([dim], [1])
+    test_data = tf.random.uniform([num_evals, batch_size, dim])  # [N, B, d]
 
     model, _, _ = trieste_deep_ensemble_model(example_data, _ENSEMBLE_SIZE)
 
-    sampler = EnsembleTrajectorySampler(model)
+    sampler = DeepEnsembleTrajectorySampler(model, diversify=diversify)
     trajectory = sampler.get_trajectory()
 
-    assert trajectory(test_data).shape == (num_evals, 1)
+    assert trajectory(test_data).shape == (num_evals, batch_size)
 
 
-def test_ensemble_trajectory_sampler_returns_deterministic_trajectory() -> None:
-    example_data = empty_dataset([1], [1])
-    test_data = tf.linspace([-10.0], [10.0], 100)
-    test_data = tf.expand_dims(test_data, -2)  # [N, 1, d]
+def test_ensemble_trajectory_sampler_returns_deterministic_trajectory(
+    num_evals: int, batch_size: int, dim: int, diversify: bool
+) -> None:
+    """
+    Evaluating the same data with the same trajectory multiple times should yield
+    exactly the same output.
+    """
+    example_data = empty_dataset([dim], [1])
+    test_data = tf.random.uniform([num_evals, batch_size, dim])  # [N, B, d]
 
     model, _, _ = trieste_deep_ensemble_model(example_data, _ENSEMBLE_SIZE)
 
-    sampler = EnsembleTrajectorySampler(model)
+    sampler = DeepEnsembleTrajectorySampler(model, diversify=diversify)
     trajectory = sampler.get_trajectory()
 
-    trajectory_eval_1 = trajectory(test_data)
-    trajectory_eval_2 = trajectory(test_data)
+    eval_1 = trajectory(test_data)
+    eval_2 = trajectory(test_data)
 
-    npt.assert_allclose(trajectory_eval_1, trajectory_eval_2)
+    npt.assert_allclose(eval_1, eval_2)
 
 
 @random_seed
 def test_ensemble_trajectory_sampler_samples_are_distinct_for_new_instances() -> None:
+    """
+    If seeds are not fixed instantiating a new sampler should give us different trajectories.
+    We use `diversify=True` here because same network could be accidentally sampled in the
+    mean mode and since we cannot fix the seed here we could get stochastic failure here.
+    """
+    diversify = True
```

Comment:
So is this test currently failing?

Also if you wanted to test it for means, you could just instantiate n samplers and check that the trajectories aren't all the same, for some n where (1/ensemble_size)^n < 0.01% or something.

Reply:
actually, with the modified sample_model_index now with means it should work without issues

Reply:
for some reason when I run just this test or `test_sampler` file it runs without an error, but when running the whole keras folder it reports an error, no idea what is going on...

Reply:
That's weird. You could maybe try commenting out other tests to see whether it's being affected by previous test state somehow. Or change the ensemble size and run again in case it's just bad luck with the RNG?